axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderLineGeneratorServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.sale.service.saleorder;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Product;
+import com.axelor.apps.base.db.repo.ProductRepository;
+import com.axelor.apps.base.db.repo.TraceBackRepository;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.sale.db.SaleOrder;
+import com.axelor.apps.sale.db.SaleOrderLine;
+import com.axelor.apps.sale.db.repo.SaleOrderLineRepository;
+import com.axelor.apps.sale.db.repo.SaleOrderRepository;
+import com.axelor.apps.sale.exception.SaleExceptionMessage;
+import com.axelor.apps.sale.service.saleorderline.SaleOrderLineComputeService;
+import com.axelor.apps.sale.service.saleorderline.SaleOrderLineDomainService;
+import com.axelor.apps.sale.service.saleorderline.SaleOrderLineInitValueService;
+import com.axelor.apps.sale.service.saleorderline.SaleOrderLineProductService;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+public class SaleOrderLineGeneratorServiceImpl implements SaleOrderLineGeneratorService {
+  protected SaleOrderLineInitValueService saleOrderLineInitValueService;
+  protected SaleOrderLineProductService saleOrderLineProductService;
+  protected SaleOrderLineRepository saleOrderLineRepository;
+  protected ProductRepository productRepository;
+  protected SaleOrderRepository saleOrderRepository;
+  protected SaleOrderLineComputeService saleOrderLineComputeService;
+  protected SaleOrderComputeService saleOrderComputeService;
+  protected SaleOrderLineDomainService saleOrderLineDomainService;
+
+  @Inject
+  public SaleOrderLineGeneratorServiceImpl(
+      SaleOrderLineInitValueService saleOrderLineInitValueService,
+      SaleOrderLineProductService saleOrderLineProductService,
+      SaleOrderRepository saleOrderRepository,
+      SaleOrderLineRepository saleOrderLineRepository,
+      SaleOrderLineComputeService saleOrderLineComputeService,
+      SaleOrderComputeService saleOrderComputeService,
+      SaleOrderLineDomainService saleOrderLineDomainService,
+      ProductRepository productRepository) {
+    this.saleOrderLineInitValueService = saleOrderLineInitValueService;
+    this.saleOrderLineProductService = saleOrderLineProductService;
+    this.saleOrderLineRepository = saleOrderLineRepository;
+    this.saleOrderRepository = saleOrderRepository;
+    this.saleOrderLineComputeService = saleOrderLineComputeService;
+    this.saleOrderComputeService = saleOrderComputeService;
+    this.saleOrderLineDomainService = saleOrderLineDomainService;
+    this.productRepository = productRepository;
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  @Override
+  public SaleOrderLine createSaleOrderLine(SaleOrder saleOrder, Product product)
```

Comment:
Add null check for SaleOrder and Product, throw exception if null, one for each.